op-node/node/conductor.go

```diff
@@ -0,0 +1,80 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-node/metrics"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/driver"
+	"github.com/ethereum-optimism/optimism/op-service/dial"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/retry"
+	"github.com/ethereum/go-ethereum/log"
+
+	conductorRpc "github.com/ethereum-optimism/optimism/op-conductor/rpc"
+)
+
+// ConductorClient is a client for the op-conductor RPC service.
+type ConductorClient struct {
+	cfg       *Config
+	apiClient conductorRpc.API
+	metrics   *metrics.Metrics
+}
+
+// NoOpConductor is a no-op conductor that assumes this node is the active sequencer.
+type NoOpConductor struct{}
+
+var _ driver.SequencerConductor = &NoOpConductor{}
+var _ driver.SequencerConductor = &ConductorClient{}
+
+// NewNoOpConductor returns a new conductor client for the op-conductor RPC service.
+func NewConductorClient(cfg *Config, log log.Logger, metrics *metrics.Metrics) (*ConductorClient, error) {
+	conductorRpcClient, err := dial.DialRPCClientWithTimeout(context.Background(), time.Minute*1, log, cfg.ConductorEndpoint())
+	if err != nil {
+		return nil, fmt.Errorf("failed to dial conductor RPC: %w", err)
+	}
+	return &ConductorClient{cfg: cfg, apiClient: conductorRpc.NewAPIClient(conductorRpcClient), metrics: metrics}, nil
+}
+
+// Active returns true if this node is the active sequencer.
+func (c *ConductorClient) Active(ctx context.Context) (bool, error) {
+	ctx, cancel := context.WithTimeout(ctx, c.cfg.ConductorRpcTimeout)
+	defer cancel()
+
+	isActive, err := retry.Do(ctx, 2, retry.Fixed(50*time.Millisecond), func() (bool, error) {
+		record := c.metrics.RecordRPCClientRequest("conductor_active")
+		result, err := c.apiClient.Active(ctx)
```

Comment:
We should probably rename this `Active` API, it returns state about the conductor itself (if it's started and not paused), instead we need to call `Leader` API to see if the node is (or should be) the active sequencer

https://github.com/ethereum-optimism/optimism/blob/1b5413b85b1f179c9d9d9d4fee05704c3c02214f/op-conductor/rpc/backend.go#L46-L48